homeassistant/components/moehlenhoff_alpha2/__init__.py

```diff
@@ -0,0 +1,124 @@
+"""Support for the Moehlenhoff Alpha2."""
+import asyncio
+import logging
+import time
+
+from moehlenhoff_alpha2 import Alpha2Base
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+class CannotConnect(exceptions.HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up Alpha2 integration."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up a config entry."""
+    base = Alpha2Base(entry.data["host"])
+    try:
+        await base.update_data()
+    except TimeoutError as err:
+        raise exceptions.ConfigEntryNotReady from err
+    base_uh = Alpha2BaseUpdateHandler(base, 60)
+    hass.data.setdefault(DOMAIN, {"connections": {}, "devices": set()})
+    hass.data[DOMAIN]["connections"][entry.entry_id] = base_uh
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok and entry.entry_id in hass.data[DOMAIN]:
+        hass.data[DOMAIN].pop(entry.entry_id)
```

Comment:
this is where the above goes wrong, the entry_id is not present under domain, but under [domain][connections], so fix this so that everything is under the entry_id, and then unpack whatever is in there.

Reply:
Done.